src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1529,6 +1530,24 @@ public function getContent($asResource = false)
         return $this->content;
     }
 
+    /**
+     * Return request body content as a json decoded array.
+     *
+     * @return mixed
+     *
+     * @throws JsonException
+     */
+    public function getJsonDecoded()
+    {
+        $json = json_decode($this->getContent(), true, 512, JSON_BIGINT_AS_STRING);
+
+        if (JSON_ERROR_NONE !== ($errorCode = json_last_error())) {
+            throw new JsonException('Unable to parse JSON: '.json_last_error_msg(), $errorCode);
```

Comment:
We could throw `\JsonException` (should be in the PHP 7.3 polyfill) here instead of inventing our own. This way we could use the `JSON_THROW_ON_ERROR` without a BC break, once Symfony does not support php 7.1/7.2 anymore.

Reply:
@derrabus , yes it is there: https://github.com/symfony/polyfill/pull/167

Reply:
Ah, I missed the part about the HTTP 400 that our custom exception should trigger. Nevermind, let's leave it like this.